### ISSUE TITLE
Update create_preassembly_spec

### ISSUE THREAD

**Issue by aaron-collier** [sul-dlss/infrastructure-integration-test#240]
With the updated manifest work now in place it looks like the integration tests needs an update:

```
❯ bundle exec rspec spec/features/create_preassembly_image_spec.rb
Run options: exclude {:indexing=>true, :skip_for_default=>true}

Randomized with seed 23168

Create and reaccession object via Pre-assembly
  is expected to match /(File image.jpg)\s\((image\/jpeg, 28.)\d( KB, preserve)\)/ (FAILED - 1)

Failures:

  1) Create and reaccession object via Pre-assembly is expected to match /(File image.jpg)\s\((image\/jpeg, 28.)\d( KB, preserve)\)/
     Failure/Error: expect(files.first.text).to match(%r{(File image.jpg)\s\((image/jpeg, 28.)\d( KB, preserve)\)})
     
       expected "File image.jpg (image/jpeg, 28.9 KB, publish/preserve)" to match /(File image.jpg)\s\((image\/jpeg, 28.)\d( KB, preserve)\)/
       Diff:
       @@ -1 +1 @@
       -/(File image.jpg)\s\((image\/jpeg, 28.)\d( KB, preserve)\)/
       +"File image.jpg (image/jpeg, 28.9 KB, publish/preserve)"
       
     # ./spec/features/create_preassembly_image_spec.rb:90:in `block (2 levels) in <top (required)>'

Finished in 45.14 seconds (files took 1.38 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/features/create_preassembly_image_spec.rb:33 # Create and reaccession object via Pre-assembly is expected to match /(File image.jpg)\s\((image\/jpeg, 28.)\d( KB, preserve)\)/

Randomized with seed 23168
```

**Comment by peetucket** [comment]
I think there may be an issue in pre-assembly, this is unexpected.  I'll have to investigate tomorrow

**Comment by peetucket** [comment]
Ok, I've been doing a lot of sleuthing, and I am now fairly certain that pre-assembly is not doing anything wrong.  It is creating `contentMetadata` without the publish/preserve/shelve attributes in this case, which is exactly what it is expected to do.

Those attributes are being added later during accessioning bythe `exif-collect` robot in `common-accessioning`, which is using defaults defined by mimetype in the `assembly-objectfile` gem to add them.  As you can see from the defined defaults below, the file used in this test has a mimetype of `image/jpeg` whose defaults are `{ preserve: 'yes', shelve: 'no', publish: 'no' }`, which is precisely what we are getting.

Bottom line: everything seems to be working as expected, so why this integration test expects something different and why it worked before still remains a mystery I am investigating.

Assembly robot where attributes are added: https://github.com/sul-dlss/common-accessioning/blob/main/lib/robots/dor_repo/assembly/exif_collect.rb#L53

The actual method that does it: https://github.com/sul-dlss/common-accessioning/blob/main/lib/dor/assembly/file_node.rb#L19-L24

The constant in `assembly-objectfile` gem that defines the default publish/preserve/shelve by mimetype: https://github.com/sul-dlss/assembly-objectfile/blob/main/lib/assembly-objectfile/content_metadata/file.rb#L11-L27

**Comment by peetucket** [comment]
Example content metadata generated by pre-assembly for an image of mimetype `image/jpeg`:
```
<?xml version="1.0"?>
<contentMetadata objectId="mn303qg4854" type="book">
  <bookData readingOrder="ltr"/>
  <resource id="mn303qg4854_1" sequence="1" type="page">
    <label>Page 1</label>
    <file id="image.jpg"/>
  </resource>
</contentMetadata>
```

After the `exif-collect` robot in `assemblyWF` is complete, you get this.  Note it has the publish/preserve/shelve and other exif data added.  There is also a new JP2 image created by the `jp2-create` robot.

```
<?xml version="1.0"?>
<contentMetadata objectId="mn303qg4854" type="book">
  <bookData readingOrder="ltr"/>
  <resource id="mn303qg4854_1" sequence="1" type="page">
    <label>Page 1</label>
    <file id="image.jpg" size="29634" mimetype="image/jpeg" preserve="yes" shelve="no" publish="no">
      <checksum type="md5">3e9498107f73ff827e718d5c743f8813</checksum>
      <checksum type="sha1">85a32f398e228e8228ad84422941110597e0d87a</checksum>
      <imageData width="500" height="700"/>
    </file>
    <file id="image.jp2" size="65743" mimetype="image/jp2" preserve="no" shelve="yes" publish="yes">
      <checksum type="md5">f354c4b11475a4c81091af1e0539cb7e</checksum>
      <checksum type="sha1">ca7ab409afcdfa5e2b924e03c957050890ee8695</checksum>
      <imageData width="500" height="700"/>
    </file>
  </resource>
</contentMetadata>
```

**Comment by peetucket** [comment]
OH...ok, I think this has nothing to do with pre-assembly or the robots.  It looks like the view of the files on argo (which is what we are checking in the test) is wrong.  `image.jpg` shows it has publish/preserve, but contentMetadata shows only preserve

![Screen Shot 2021-04-23 at 1 37 05 PM](https://user-images.githubusercontent.com/47137/115927285-0a6def00-a439-11eb-8b9f-f9928c2a5b3e.png)
![Screen Shot 2021-04-23 at 1 37 11 PM](https://user-images.githubusercontent.com/47137/115927294-0cd04900-a439-11eb-8ddf-82e335d17479.png)

**Comment by peetucket** [comment]
And here is the problem in argo:

https://github.com/sul-dlss/argo/blob/main/app/components/contents/file_component.rb#L25

It shows "publish" for any "world" object regardless of what the actual file sets the publish attribute to.

So... that explains everything except why the integration test hasn't been failing since that commit was made (1 month ago)

**Comment by peetucket** [comment]
PR to fix this in argo: sul-dlss/argo#2708

**Comment by ndushay** [comment]
Can this be closed?  sul-dlss/argo#2708 has been merged.

**Comment by peetucket** [comment]
Yup, I just confirmed it works now on stage.
